engine/src/main/java/org/terasology/logic/inventory/ItemPickupAuthoritySystem.java

```diff
@@ -59,20 +66,32 @@ public void onDropItemEvent(DropItemEvent event, EntityRef itemEntity, ItemCompo
         if (!itemEntity.hasComponent(LocationComponent.class)) {
             itemEntity.addComponent(new LocationComponent(event.getPosition()));
         }
+
+        itemEntity.getComponent(ItemComponent.class).pickableUp = false;
+        long timeToPickUp = itemEntity.getComponent(ItemComponent.class).pickUpTime;
+        delayManager.addDelayedAction(itemEntity, DELAYED_PICKUP_ACTION_ID, timeToPickUp);
     }
 
+    @ReceiveEvent
+    public void onDelayedPickup(DelayedActionTriggeredEvent event, EntityRef entityRef) {
```

Comment:
Good point (both about the optimization and the fact that I forgot checks... all over the place ^^' )